target_chains/ethereum/contracts/contracts/pyth/Pyth.sol

```diff
@@ -432,84 +432,144 @@ abstract contract Pyth is
+                            memory accumulatorPriceInfos,
+                        bytes32[] memory accumulatorPriceIds
+                    ) = parsePricesUsingAccumulator(updateData[i]);
+
+                    for (
+                        uint accDataIdx = 0;
+                        accDataIdx < accumulatorPriceIds.length;
+                        accDataIdx++
+                    ) {
+                        bytes32 accumulatorPriceId = accumulatorPriceIds[
+                            accDataIdx
+                        ];
+                        // check whether caller requested for this data
+                        uint k = getPriceFeedsIndex(
+                            priceIds,
+                            accumulatorPriceId
+                        );
 
-                (
-                    uint index,
-                    uint nAttestations,
-                    uint attestationSize
-                ) = parseBatchAttestationHeader(encoded);
-
-                // Deserialize each attestation
-                for (uint j = 0; j < nAttestations; j++) {
-                    // NOTE: We don't advance the global index immediately.
-                    // attestationIndex is an attestation-local offset used
-                    // for readability and easier debugging.
-                    uint attestationIndex = 0;
-
-                    // Unused bytes32 product id
-                    attestationIndex += 32;
-
-                    bytes32 priceId = UnsafeBytesLib.toBytes32(
-                        encoded,
-                        index + attestationIndex
-                    );
+                        // If priceFeed[k].id != 0 then it means that there was a valid
+                        // update for priceIds[k] and we don't need to process this one.
+                        if (k == priceIds.length || priceFeeds[k].id != 0) {
+                            continue;
+                        }
 
-                    // Check whether the caller requested for this data.
-                    uint k = 0;
-                    for (; k < priceIds.length; k++) {
-                        if (priceIds[k] == priceId) {
-                            break;
+                        PythInternalStructs.PriceInfo
+                            memory info = accumulatorPriceInfos[accDataIdx];
+
+                        uint publishTime = uint(info.publishTime);
+                        // Check the publish time of the price is within the given range
+                        // if it is not, then set the id to 0 to indicate that this price id
+                        // still does not have a valid price feed. This will allow other updates
+                        // for this price id to be processed.
+                        if (
+                            publishTime < minPublishTime ||
+                            publishTime > maxPublishTime
+                        ) {
+                            priceFeeds[k].id = 0;
```

Comment:
isn't this guaranteed to be 0 at this point (because line 463 implies this is zero, and you haven't populated it after that)?

Reply:
I agree, in the old code we did the deserialization in place (for gas saving) and that's why we needed a reset here but here we don't need it.

Reply:
yes that's correct thanks